src/manifest/errors.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  ManifestCustomEventSchema,
+  ManifestCustomTypeSchema,
+  ManifestDatastoreSchema,
+  ManifestFunctionSchema,
+  ManifestOAuth2ProviderSchema,
+  ManifestWorkflowSchema,
+} from "./manifest_schema.ts";
+
+export class DuplicateWorkflowError extends Error {
+  static stringifySimpleSchema = (schema: ManifestWorkflowSchema) =>
+    JSON.stringify(
+      {
+        title: schema.title ?? "",
+        description: schema.description ?? "",
+      },
+      null,
+      1,
+    );
+
+  constructor(
+    id: string,
+    current: ManifestWorkflowSchema,
+    old: ManifestWorkflowSchema,
+  ) {
+    super([
+      `Duplicate callback_id: "${id}" for Workflow`,
+      DuplicateWorkflowError.stringifySimpleSchema(current),
+      "---",
+      DuplicateWorkflowError.stringifySimpleSchema(old),
+    ].join("\n"));
```

Comment:
The output is a little weird:

```
error: Uncaught (in promise) Error: Duplicate callback_id: "create_new_issue_workflow" for Workflow
{
"title": "Create new issue2",
"description": "Create a new GitHub issue"
}
---
{
"title": "Create new issue",
"description": "Create a new GitHub issue"
}
```

Do we have to show the error data as JSON? And I think displaying title and description is not ideal, as I often duplicate these fields (and they don't need to be unique, either). Can we instead display the path to the relevant source files containing the duplicate IDs?

Reply:
We do not need to show the error data as JSON, I opted for this solution since I felt like like it provided the data in a clean way, maybe a table pattern is better

> Can we instead display the path to the relevant source files containing the duplicate IDs?

Figuring out what data to display to the user in order to trace down what workflows are duplicated has proven to be a tricky one, for functions it is easy to display the `source_file` field but for other assets like workflows or customTypes they don't have this field and could be defined anywhere in the project

I opted to display the `title` and `description` because it provides some information to the user, my logic was that if a user copy and pastes a workflow and forget to change the `callback_id`, `title` and `description`, displaying the  `title` and `description` might remind them of which workflow they copy and pasted

@filmaj what did you have in mind to fetch the source_files of the objects?

Reply:
> maybe a table pattern is better

I don't think we need to show diffs, we just need to tell the developer where the duplicate IDs are so that they can fix the problem.

> Figuring out what data to display to the user in order to trace down what workflows are duplicated has proven to be a tricky one

I see... yes good point. That kinda sux. I wonder if there's something we can do (future potential improvement / research) where the `Define*` functions can use the various module metadata APIs deno provides in order to figure out the source location. Maybe each `*Definition` object logs its `import.meta.url` in a private property, the main `Manifest` object does the same, and then some of the `Manifest` methods can use `import.meta.resolve()` between the Manifest's URL and the various `*Definition` URLs to figure out source locations?

Reply:
> I wonder if there's something we can do (future potential improvement / research)

Did some research on this earlier seems like its might be a complicated solution, I will dumb down this PR and ask for a second round of review